mypy/build.py

```diff
@@ -933,13 +937,28 @@ def _find_module(self, id: str, lib_path: Tuple[str, ...],
             typed_file = os.path.join(pkg_dir, components[0], 'py.typed')
             stub_dir = os.path.join(pkg_dir, stub_name)
             if fscache.isdir(stub_dir):
+                stub_typed_file = os.path.join(stub_dir, 'py.typed')
                 stub_components = [stub_name] + components[1:]
                 path = os.path.join(pkg_dir, *stub_components[:-1])
                 if fscache.isdir(path):
-                    third_party_stubs_dirs.append(path)
+                    if fscache.isfile(stub_typed_file):
+                        # Stub packages can have a py.typed file, which must include
+                        # 'partial\n' to make the package partial
+                        # Partial here means that mypy should look at the runtime
+                        # package if installed.
+                        with open(stub_typed_file) as f:
+                            src = f.read()
+                        if src == 'partial\n':
```

Comment:
Although the PEP does explicitly say `partial\n`, would it be better to just check `src.strip() == 'partial'`? It doesn't seem worth worrying about the exact newline representation.

Reply:
Ah, I should probably use `fscache.read()` here